python-pytest.el

```diff
@@ -615,7 +700,9 @@ Example: ‘MyABCThingy.__repr__’ becomes ‘test_my_abc_thingy_repr’."
     ;; check all project buffers
     (-when-let*
         ((buffers
-          (projectile-buffers-with-file (projectile-project-buffers)))
+          (if (python-pytest--using-projectile)
+              (projectile-buffers-with-file (projectile-project-buffers))
+            (project-buffers (project-current t))))
          (modified-buffers
           (-filter 'buffer-modified-p buffers))
```

Comment:
```suggestion
          (-filter (lambda (x) (and (buffer-modified-p x) (buffer-file-name x))) buffers))
```
Currently this attempts to save all buffers, even non-file-visiting ones like eshell and eglot buffers.

Maybe something like `magit-save-repository-buffers` could be useful in giving a more interactive prompt

Reply:
Actually, in a `projectile` such a filter has already been applied, see `projectile-buffers-with-file`. I'd suggest to make the `project` branch equivalent to `projectile` one, i.e update the `ELSE` branch above to:
```
          (if (python-pytest--using-projectile)
              (projectile-buffers-with-file (projectile-project-buffers))
            (cl-remove-if-not (lambda (b)
                                (buffer-file-name b)) 
                              (project-buffers (project-current t))))
```

Reply:
Great find and feedback. Thanks!

Updated to not include non-file-backed buffers.  And noticed while testing that, if file deleted that was in recentf list would fail.